### PR TITLE
Add workflow example for time series forecasting

### PR DESCRIPTION
**Pull request by hcho3**
* Use data from the M5 competition, a well-known competition for time series forecasting
* Use Dask Kubernetes to run hyperparameter optimization
* Use GPU-accelerated XGBoost and cuDF to accelerate time series analysis end-to-end
* All preprocessing logic uses cuDF.

### CODE REVIEW

File: source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb [hunk 1]
```diff
@@ -0,0 +1,3201 @@
+{
```
Comment (trivialfis) [inline source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    import pandas as pd
</pre><p>Where is pandas being used?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: source/examples/time-series-forecasting-with-hpo/preprocessing_part2.ipynb [hunk 2]
```diff
@@ -0,0 +1,1259 @@
+{
```
[the remark this answers is not in the document]
Comment (hcho3) [reply source/examples/time-series-forecasting-with-hpo/preprocessing_part2.ipynb:1]: <p>Two reasons:</p><ol><li><code>cudf.DataFrame.to_parquet</code> doesn't yet support writing categorical columns.</li><li><code>cudf.DataFrame.to_parquet</code> doesn't yet support writing to a file stream. This is necessary for writing to Google Cloud Storage.</li></ol>

File: source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb (hunk 1 above)
Comment (hcho3) [reply source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb:1]: <p>Removed.</p>

**Comment by hcho3** [comment]
@jacobtomlinson For the time being, I put in the extra notebooks under the `toctree` of the `start_here.ipynb`. I did so because I didn't want to clutter the workflow catalog.

File: source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb [hunk 3]
```diff
@@ -0,0 +1,3219 @@
...
```
Comment (jacobtomlinson) [inline source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb:1]: <p>Ooh I'd not seen <code>%xdel</code> before, neat! </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (jacobtomlinson) [inline source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb:1]: <p>Given that you are calling <code>to_pandas</code> here I wonder if you could use <code>to_parquet</code> to write to GCS?  Or is the categorical column still a problem?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

File: source/examples/time-series-forecasting-with-hpo/start_here.ipynb [hunk 4]
```diff
@@ -0,0 +1,309 @@
+{
```
Comment (jacobtomlinson) [inline source/examples/time-series-forecasting-with-hpo/start_here.ipynb:1]: <p>Can you use a <code>docref</code> directive here to show this is a reference to another documentation page?</p><p> </p><p>See this example https://github.com/rapidsai/deployment/blob/6c81beac81158ceae58ded3b273e8e4c440047cd/source/examples/rapids-azureml-hpo/notebook.ipynb?short_path=4cb4321#L27-L32 </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: source/examples/time-series-forecasting-with-hpo/training_and_evaluation.ipynb [hunk 5]
```diff
@@ -0,0 +1,1777 @@
+{
```
Comment (jacobtomlinson) [inline source/examples/time-series-forecasting-with-hpo/training_and_evaluation.ipynb:1]: <p>These maths expressions are not rendering correctly in the Sphinx docs. We might need to enable <code>dollarmath</code> in our <code>conf.py</code> https://myst-parser.readthedocs.io/en/latest/syntax/optional.html.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (jacobtomlinson) [inline source/examples/time-series-forecasting-with-hpo/training_and_evaluation.ipynb:1]: <p>It feels like we end abruptly here. Could you add a conclusion?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

**Review by jacobtomlinson — COMMENTED** [review]
Thanks @hcho3 sorry it has taken me so long to get a review on this one.

Overall I think the structure is great, I've added a few comments.

One high-level question I had is why are there so many separate pre-processing notebooks? Some of them seem to load the data, do some small transformation and save them again. I think it would be more pleasant to read if there were fewer (or just one) pre-processing notebook. What do you think?

File: source/examples/time-series-forecasting-with-hpo/training_and_evaluation.ipynb (hunk 5 above)
Comment (jacobtomlinson) [reply source/examples/time-series-forecasting-with-hpo/training_and_evaluation.ipynb:1]: I've pushed a commit to resolve this

File: source/examples/time-series-forecasting-with-hpo/start_here.ipynb (hunk 4 above)
Comment (jacobtomlinson) [inline source/examples/time-series-forecasting-with-hpo/start_here.ipynb:1]: <p>We might want to move the toctree up here to make it clear this is a multi-page example.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/255/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb (hunk 3 above)
Comment (hcho3) [reply source/examples/time-series-forecasting-with-hpo/preprocessing_part1.ipynb:1]: <p>Yes, you cannot save DataFrame to Parquet if it contains categorical columns.</p>

File: source/examples/time-series-forecasting-with-hpo/start_here.ipynb (hunk 4 above)
Comment (hcho3) [reply source/examples/time-series-forecasting-with-hpo/start_here.ipynb:1]: <p>I combined all notebooks to a single notebook, so no need for ToC anymore.</p>

**Review by jacobtomlinson — COMMENTED** [review]
This is looking really great. Thanks for taking the time to merge this into one notebook and 
address the review feedback.

The only small nit I have is that the `DISABLE_JUPYTER` env var is no longer needed, so you can remove that from the code cell.

**Pull request merged**